exercises/practice/two-bucket/src/test/java/TwoBucketTest.java

```diff
@@ -75,4 +76,52 @@ public void testBucketOneSizeTwoBucketTwoSizeThreeStartWithOne() {
         assertThat(twoBucket.getOtherBucket()).isEqualTo(2);
 
     }
+
+    @Disabled("Remove to run test")
+    @Test
+    public void testReachingGoalIsImpossible() {
+
+        TwoBucket twoBucket = new TwoBucket(6, 15, 5, "one");
+
+        assertThatExceptionOfType(IllegalArgumentException.class)
```

Comment:
I think it is a bit odd to have to have to assert that we get an exception three times in the test (I was expecting only one assertion). What do you think about refactoring to assert the exception just once? In the [csharp](https://github.com/exercism/csharp/blob/main/exercises/practice/two-bucket/TwoBucketTests.cs) implementation, the `TwoBucket` class has a `measure` method that the tests calls. It returns a "Result" object containing the number of moves, the final bucket and the other bucket or throws an exception if the goal can be reached.

Reply:
> I think it is a bit odd to have to have to assert that we get an exception three times in the test

Even I found it a bit odd to assert the exception three times while implementing the tests but couldn’t think of a better alternative. However, your suggestion of returning a `Result` object and handling everything within a single method is an excellent idea! This approach not only simplifies the implementation but also ensures the custom exception can be thrown effectively.